packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -218,13 +263,215 @@ export const Tabs = (props: TabsProps) => {
+      return -1; // tab is before visible section
+    } else {
+      return 1; // tab is after visible section
     }
-    setCurrentActiveIndex(index);
   };
 
+  function easeInOutQuad(
+    time: number,
+    beginning: number,
+    change: number,
+    duration: number,
+  ) {
+    if ((time /= duration / 2) < 1) {
+      return (change / 2) * time * time + beginning;
+    }
+    return (-change / 2) * (--time * (time - 2) - 1) + beginning;
+  }
+
+  const scrollToTab = React.useCallback(
+    (
+      list: HTMLUListElement,
+      activeTab: HTMLElement,
+      duration: number,
+      isVertical: boolean,
+      tabPlacement: number,
+    ) => {
+      const start = isVertical ? list.scrollTop : list.scrollLeft;
+      let change = 0;
+      let currentTime = 0;
+      const increment = 20;
+      const fadeBuffer = isVertical
+        ? list.offsetHeight * 0.05
+        : list.offsetWidth * 0.05;
+
+      if (tabPlacement < 0) {
+        // if tab is before visible section
+        change = isVertical
+          ? activeTab.offsetTop - list.scrollTop
+          : activeTab.offsetLeft - list.scrollLeft;
+        change -= fadeBuffer; // give some space so the active tab isn't covered by the fade
+      } else {
+        // tab is after visible section
+        change = isVertical
+          ? activeTab.offsetTop -
+            (list.scrollTop + list.offsetHeight) +
+            activeTab.offsetHeight
+          : activeTab.offsetLeft -
+            (list.scrollLeft + list.offsetWidth) +
+            activeTab.offsetWidth;
+        change += fadeBuffer; // give some space so the active tab isn't covered by the fade
+      }
+
+      const animateScroll = () => {
+        currentTime += increment;
+        const val = easeInOutQuad(currentTime, start, change, duration);
+        if (isVertical) {
+          list.scrollTop = val;
+        } else {
+          list.scrollLeft = val;
+        }
+        if (currentTime < duration) {
+          setTimeout(animateScroll, increment);
+        }
+      };
+      animateScroll();
+    },
+    [],
+  );
+
+  // scroll to active tab if it is not visible with overflow
+  useIsomorphicLayoutEffect(() => {
+    setTimeout(() => {
+      const ownerDoc = tablistRef.current;
+      if (
+        ownerDoc !== null &&
+        overflowOptions?.useOverflow &&
+        currentActiveIndex !== undefined
+      ) {
+        const activeTab = ownerDoc.querySelectorAll('.iui-tab')[
+          currentActiveIndex
+        ] as HTMLElement;
+        const isVertical = orientation === 'vertical';
+        const tabPlacement = isTabHidden(activeTab, isVertical);
+
+        if (tabPlacement) {
+          scrollToTab(ownerDoc, activeTab, 100, isVertical, tabPlacement);
+        }
+      }
+    }, 50);
+  }, [
+    overflowOptions?.useOverflow,
+    currentActiveIndex,
+    focusedIndex,
+    orientation,
+    scrollToTab,
+  ]);
+
+  const [scrollingPlacement, setScrollingPlacement] = React.useState('none');
```

Comment:
Having 'none' initial value adds this prop to ALL tabs, even not overflowing ones. Change to undefined.
```suggestion
  const [scrollingPlacement, setScrollingPlacement] = React.useState('none');  const [scrollingPlacement, setScrollingPlacement] = React.useState<
    string | undefined
  >(undefined);
```